tenpy/algorithms/dmrg.py

```diff
@@ -838,101 +676,550 @@ def mixed_svd(self, theta, i0, update_LP, update_RP):
         err : :class:`~tenpy.algorithms.truncation.TruncationError`
             The truncation error introduced.
         """
+        i0 = self.i0
         # get qtotal_LR from i0
         if self.mixer is None:
             # simple case: real svd, defined elsewhere.
-            qtotal_i0 = self.psi.get_B(i0, form=None).qtotal
+            qtotal_i0 = self.env.bra.get_B(i0, form=None).qtotal
             U, S, VH, err, _ = svd_theta(theta,
                                          self.trunc_params,
                                          qtotal_LR=[qtotal_i0, None],
                                          inner_labels=['vR', 'vL'])
             return U, S, VH, err
         # else: we have a mixer
-        return self.mixer.perturb_svd(self, theta, i0, update_LP, update_RP)
+        if self.finite:
+            return self.mixer.perturb_svd(self, theta, self.i0, self.move_right, (not self.move_right))
```

Comment:
I'm very much confused by why this is necessary. At which point in the finite sweep is `move_right` different from `update_LP` or `not move_right` different from `update_RP`???

Reply:
Resolved in 9bd09b0eb1cc40bb9d57ae61430fcdf853e4a11b.

This one will make you laugh: we were calling on self.update_LP and self.update_RP, expecting these to be the bools set by the schedule. However, the bools are in self.update_LP_RP; while update_LP/*_RP are class methods...